tests/test_migrate.py

```
@@ -0,0 +1,1792 @@
+        creds = legacy_root / "credentials.json"
+        creds.write_text("{}")
+        new_home = tmp_path / "home"
+        new_home.mkdir()
+
+        with patch("shutil.copy2", side_effect=OSError("Mock error")):
+            result = migrate_credentials([legacy_root], new_home, move=False)
+
+            assert result["success"] is False
+            assert "Mock error" in result["error"]
+
+    def test_backup_oserror_logged(self, tmp_path, monkeypatch):
```

Ruff (0.14.14)

[warning] Unused method argument: `monkeypatch`

(ARG002)